pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.operator.filter;
+
+import com.linkedin.pinot.core.common.DataSource;
+import com.linkedin.pinot.core.common.DataSourceMetadata;
+import com.linkedin.pinot.core.common.Predicate;
+import com.linkedin.pinot.core.operator.filter.predicate.PredicateEvaluator;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+
+public class FilterOperatorUtils {
```

Comment:
These methods are common methods which are used to build the filter operator. They are used in both filter plan node and start tree operator.

Reply:
These methods are common filter operator related methods which are used in both plan node and star tree operator. Calling methods in plan node from operator will cause some confusion.

Reply:
Lets created separate classes for each method. Its ok to have just one method in the class than having a Utils class with a bunch of methods with complex logic.

Reply:
I will file another PR to change the starTreeOperator to a filter plan node (which make more sense because actually it just creates another filter operator), and move these methods into a base class.